src/script/api/script_text.cpp

```diff
@@ -220,46 +229,54 @@ void ScriptText::_GetEncodedText(std::back_insert_iterator<std::string> &output,
 	auto skip_args = [&](size_t nb) { idx += nb; };
 
 	for (const StringParam &cur_param : params) {
-		switch (cur_param.type) {
-			case StringParam::UNUSED:
-				skip_args(cur_param.consumes);
-				break;
-
-			case StringParam::RAW_STRING: {
-				ParamCheck &p = *get_next_arg();
-				p.Encode(output);
-				if (!std::holds_alternative<std::string>(*p.param)) ScriptLog::Error(fmt::format("{}({}): {{{}}} expects a raw string", name, param_count + 1, cur_param.cmd));
-				break;
-			}
+		try {
+			switch (cur_param.type) {
+				case StringParam::UNUSED:
+					skip_args(cur_param.consumes);
+					break;
 
-			case StringParam::STRING: {
-				ParamCheck &p = *get_next_arg();
-				p.Encode(output);
-				if (!std::holds_alternative<ScriptTextRef>(*p.param)){
-					ScriptLog::Error(fmt::format("{}({}): {{{}}} expects a GSText", name, param_count + 1, cur_param.cmd));
+				case StringParam::RAW_STRING:
+				{
+					ParamCheck &p = *get_next_arg();
+					p.Encode(output, cur_param.cmd);
+					if (!std::holds_alternative<std::string>(*p.param)) ScriptLog::Error(fmt::format("{}({}): {{{}}} expects a raw string", name, param_count + 1, cur_param.cmd));
 					break;
 				}
-				int count = 0;
-				ScriptTextRef &ref = std::get<ScriptTextRef>(*p.param);
-				ref->_GetEncodedText(output, count, args.subspan(idx), false);
-				if (++count != cur_param.consumes) {
-					ScriptLog::Error(fmt::format("{}({}): {{{}}} expects {} to be consumed, but {} consumes {}", name, param_count + 1, cur_param.cmd, cur_param.consumes - 1, GetGameStringName(ref->string), count - 1));
-					/* Fill missing params if needed. */
-					for (int i = count; i < cur_param.consumes; i++) fmt::format_to(output, ":0");
-				}
-				skip_args(cur_param.consumes - 1);
-				break;
-			}
 
-			default:
-				for (int i = 0; i < cur_param.consumes; i++) {
+				case StringParam::STRING:
+				{
 					ParamCheck &p = *get_next_arg();
-					p.Encode(output);
-					if (!std::holds_alternative<SQInteger>(*p.param)) ScriptLog::Error(fmt::format("{}({}): {{{}}} expects an integer", name, param_count + i + 1, cur_param.cmd));
+					p.Encode(output, cur_param.cmd);
+					if (p.cmd != cur_param.cmd) throw 1;
+					if (!std::holds_alternative<ScriptTextRef>(*p.param)) {
+						ScriptLog::Error(fmt::format("{}({}): {{{}}} expects a GSText", name, param_count + 1, cur_param.cmd));
+						throw 1;
```

Comment:
Why not something like
```
param_count++;
continue;
```
instead?

I'm seeing only one throw, and in this case you just want to add something to `param_count` and then continue with the next parameter, right?

Reply:
There are more than one throw, but changing this one allows me to put a warning in the `catch`.